File: src/leetcode/2026/05/Solution977.java
public class Solution977 {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length;
        int[] result = new int[n];
        int left = 0, right = n - 1;
        int pos = n - 1;

        while (left <= right) {
            if (Math.abs(nums[left]) > Math.abs(nums[right])) {
                result[pos] = nums[left] * nums[left];
                left++;
            } else {
                result[pos] = nums[right] * nums[right];
                right--;
            }
            pos--;
        }

        return result;
    }

    public static void main(String[] args) {
        Solution977 solution = new Solution977();

        int[] nums1 = { -4, -1, 0, 3, 10 };
        int[] result1 = solution.sortedSquares(nums1);
        System.out.print("Output for nums1: ");
        for (int num : result1) {
            System.out.print(num + " ");
        }
        System.out.println(); // Output: [0, 1, 9, 16, 100]

        int[] nums2 = { -7, -3, 2, 3, 11 };
        int[] result2 = solution.sortedSquares(nums2);
        System.out.print("Output for nums2: ");
        for (int num : result2) {
            System.out.print(num + " ");
        }
        System.out.println(); // Output: [4, 9, 9, 49, 121]
    }
}

File: src/leetcode/2026/05/Solution977.cpp
#include <vector>
#include <iostream>
using namespace std;

class Solution977 {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int n = nums.size();
        vector<int> result(n);
        int left = 0, right = n - 1, index = n - 1;

        while (left <= right) {
            int leftSquare = nums[left] * nums[left];
            int rightSquare = nums[right] * nums[right];
            if (leftSquare > rightSquare) {
                result[index--] = leftSquare;
                left++;
            } else {
                result[index--] = rightSquare;
                right--;
            }
        }
        return result;
    }
};

int main() {
    Solution977 solution;

    vector<int> nums1 = {-4, -1, 0, 3, 10};
    vector<int> result1 = solution.sortedSquares(nums1);
    for (int num : result1) {
        cout << num << " ";
    }
    cout << endl;

    vector<int> nums2 = {-7, -3, 2, 3, 11};
    vector<int> result2 = solution.sortedSquares(nums2);
    for (int num : result2) {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}